datafusion/src/optimizer/constant_folding.rs

```diff
@@ -217,6 +218,29 @@ impl<'a> ExprRewriter for ConstantRewriter<'a> {
                     .query_execution_start_time
                     .timestamp_nanos(),
             ))),
+            Expr::ScalarFunction {
```

Comment:
Matching on the literal argument is good as the `rewrite` pass happens *after* all children are visited 👍